models/BaseEntity.cfc

```diff
@@ -2380,14 +2455,51 @@ component accessors="true" {
 		if ( variables._meta.originalMetadata.keyExists( "grammar" ) ) {
 			variables._builder.setGrammar( variables._wirebox.getInstance( variables._meta.originalMetadata.grammar ) );
 		}
-		retrieveQuery().from( tableName() );
-		return variables._builder
+
+		variables._builder
 			.setEntity( this )
 			.setReturnFormat( "array" )
 			.setDefaultOptions( variables._queryOptions )
-			.from( tableName() )
-			.newQuery()
-			.select( retrieveQualifiedColumns() );
+			.from( tableName() );
+
+		if ( variables._hasParentEntity ) {
+			return variables._builder
+				.newQuery()
+				.select( retrieveQualifiedColumns() )
+				.join(
+					variables._parentEntity.meta.table,
+					variables._parentEntity.meta.table & "." & variables._parentEntity.key,
+					this.qualifyColumn( variables._key )
+				);
+		} else if (
+			variables._meta.keyExists( "discriminatorColumn" ) && application.quickMeta.discriminators.keyExists(
```

Comment:
The way I understand this is

If this entity has a discriminator column
and if we know of a parent entity for this entity's table,
then do stuff....

But what if we have a `discriminatorColumn` and we _don't_ know of a parent entity for this entity's table?  Shouldn't that be an exception with a helpful error message?

Reply:
In fact `joincolumn`, `discriminatorColumn`, `discriminatorValue`, and `extends` are all required if one is provided.  I think we can throw if that's not the case when inspecting the metadata with a helpful explanation on how to fix it.

Reply:
`discriminatorColumn` is always on the parent ( see above ).  `discriminatorValue` is always on the child, as is `extends` and `joinColumn`.  

Since the application discriminator data is detected when inspecting the _children_, if the key is not in the application scope, there are no registered children and we would fall through to to the normal behavior of an entity without parents or children.